services/checks/checks.go

```diff
@@ -565,46 +570,75 @@ type sttCheckResult struct {
 	result    check.Result
 }
 
-func (s *Service) processResults(ctx context.Context, check check.Check, target target, resID string) ([]sttCheckResult, error) {
+// StarlarkScriptData represents the data we need to pass to the binary to run starlark scripts.
+type StarlarkScriptData struct {
+	CheckName    string                   `json:"name"`
+	Script       string                   `json:"script"`
+	CheckVersion uint32                   `json:"version"`
+	ScriptInput  []map[string]interface{} `json:"input"`
+}
+
+func (s *Service) processResults(ctx context.Context, sttCheck check.Check, target target, resID string) ([]sttCheckResult, error) {
 	nCtx, cancel := context.WithTimeout(ctx, resultTimeout)
 	r, err := s.waitForResult(nCtx, resID)
 	cancel()
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to get action result")
 	}
 
-	funcs, err := getFuncsForVersion(check.Version)
+	l := s.l.WithFields(logrus.Fields{
+		"name":       sttCheck.Name,
+		"id":         resID,
+		"service_id": target.serviceID,
+	})
+
+	input := &StarlarkScriptData{
+		CheckName:    sttCheck.Name,
+		Script:       sttCheck.Script,
+		CheckVersion: sttCheck.Version,
+		ScriptInput:  r,
+	}
+
+	cmdCtx, cancel := context.WithTimeout(ctx, scriptTimeout)
+	defer cancel()
+
+	process := exec.CommandContext(cmdCtx, "pmm-managed-starlark")
+	pdeathsig.Set(process, syscall.SIGTERM)
+
+	pipe, err := process.StdinPipe()
+	defer pipe.Close()
+	if err != nil {
+		return nil, err
+	}
+
+	encoder := json.NewEncoder(pipe)
+	err = encoder.Encode(input)
 	if err != nil {
+		l.Error("Error encoding data to STDIN, ", err)
 		return nil, err
 	}
 
-	env, err := starlark.NewEnv(check.Name, check.Script, funcs)
+	procOut, err := process.CombinedOutput()
```

Comment:
It should not be combined output – we need only stdout. Stderr should be captured as usual.